internal/cli/cmd/cluster/git.go

```diff
@@ -0,0 +1,73 @@
+package cluster
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+type gitScheme string
+
+const (
+	SSH   gitScheme = "ssh"
+	HTTPS gitScheme = "https"
+)
+
+type giturl struct {
+	scheme                      gitScheme
+	hostname, owner, repository string
+}
+
+func (g giturl) URL() string {
+	if g.scheme == SSH {
+		return fmt.Sprintf("git@%s:%s/%s.git", g.hostname, g.owner, g.repository)
+	} else {
+		return fmt.Sprintf("https://%s/%s/%s.git", g.hostname, g.owner, g.repository)
+	}
+}
+
+func parseGitUrl(url string) (giturl, error) {
```

Comment:
Probably not relevant in this particular case but I think file:// also exists (which can be implied by being left out) and git://
https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols